lib/src/merge.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright 2023 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::hash::Hash;
+
+use itertools::Itertools;
+
+/// Attempt to resolve trivial conflicts between the inputs. Even indexes are
+/// positive terms, and odd indexes are negative terms. The length must be odd.
+pub fn trivial_merge<T>(input: &[T]) -> Option<T>
+where
+    T: Eq + Hash + Clone,
+{
+    assert_eq!(input.len() % 2, 1);
```

Comment:
I would add a message here. 
```rust
assert_eq!(input.len() % 2, 1, "inputs must be odd");
```

Reply:
Done. At first I was thinking it would not be useful to users anyway, and developers will look up the line of code anyway. But I suppose it could help as a way of identifying which assertion it is without knowing exactly which version of the code a user was running.